aws/middleware/header.go

```diff
@@ -0,0 +1,82 @@
+package middleware
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/aws/smithy-go/middleware"
+	smithyhttp "github.com/aws/smithy-go/transport/http"
+)
+
+// WithHeaderComment instruments a middleware stack to append an HTTP field
+// comment to the given header as specified in RFC 9110
+// (https://www.rfc-editor.org/rfc/rfc9110#name-comments).
+//
+// The header is case-insensitive. If the provided header exists when the
+// middleware runs, the content will be inserted as-is enclosed in parentheses.
+//
+// Note that per the HTTP specification, comments are only allowed in fields
+// containing "comment" as part of their field value definition, but this API
+// will NOT verify whether the provided header is one of them.
+//
+// WithHeaderComment MAY be applied more than once to a middleware stack and/or
+// more than once per header.
+func WithHeaderComment(header, content string) func(*middleware.Stack) error {
```

Comment:
**nit**: Seems like it could have existed in `smithy-go` since it's not really tied to AWS

Reply:
^agree with the above. i think this is prob more than a `nit` given that its an exported function, and we cant ever move it later (this is my only issue, and fine with approving after a response or fixed).

Reply:
I think my main objection to doing so is that `smithy-go/middleware` doesn't actually _implement_ any middleware, it's just the middleware definition itself. Shoehorning this extra API, generic as it may be, doesn't feel quite right organizationally.

Reply:
@lucix-aws would it fit anywhere else in smithy-go? if not, and you still think it fits organizationally better where it is -- then im fine with that and feel free to merge